rocketship/dce/models.py

```diff
@@ -0,0 +1,390 @@
+"""
+DCE-MRI models for ROCKETSHIP.
+
+Pharmacokinetic models for DCE-MRI analysis including Tofts, Extended Tofts, Patlak, etc.
+"""
+
+import numpy as np
+from scipy import optimize, integrate, interpolate
+from typing import Tuple, Optional, Dict, Any, Callable
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class DCEModel:
+    """Base class for DCE-MRI pharmacokinetic models."""
+    
+    def __init__(self, name: str):
+        self.name = name
+        self.parameter_names = []
+        self.parameter_bounds = {}
+        self.initial_values = {}
+    
+    def model_function(self, t: np.ndarray, *params) -> np.ndarray:
+        """Model function - must be implemented by subclasses."""
+        raise NotImplementedError
+    
+    def fit(self, t: np.ndarray, ct: np.ndarray, cp: np.ndarray, **kwargs) -> Dict[str, Any]:
+        """Fit model to data."""
+        raise NotImplementedError
+
+
+class ToftsModel(DCEModel):
+    """
+    Standard Tofts model for DCE-MRI.
+    
+    Ct(t) = Ktrans * Cp(t) ⊗ exp(-Ktrans*t/ve)
+    
+    Parameters:
+        - Ktrans: volume transfer constant
+        - ve: extravascular extracellular space volume fraction
+    """
+    
+    def __init__(self):
+        super().__init__("Tofts")
+        self.parameter_names = ['Ktrans', 've']
+        self.parameter_bounds = {
+            'Ktrans': (1e-7, 2.0),
+            've': (0.02, 1.0)
+        }
+        self.initial_values = {
+            'Ktrans': 0.0002,
+            've': 0.2
+        }
+    
+    def model_function(self, t: np.ndarray, ktrans: float, ve: float, cp: np.ndarray) -> np.ndarray:
+        """
+        Tofts model function.
+        
+        Args:
+            t: Time points
+            ktrans: Volume transfer constant
+            ve: Extravascular extracellular volume fraction
+            cp: Arterial input function
+            
+        Returns:
+            Modeled concentration
+        """
+        if ktrans <= 0 or ve <= 0:
+            return np.zeros_like(t)
+        
+        # Convolution with exponential decay
+        kep = ktrans / ve
+        exp_decay = np.exp(-kep * t)
+        
+        # Convolution using numerical integration
+        ct = np.zeros_like(t)
+        dt = np.mean(np.diff(t)) if len(t) > 1 else 1.0
+        
+        for i in range(len(t)):
+            if i == 0:
+                ct[i] = 0
+            else:
+                # Trapezoidal integration
+                integrand = cp[:i+1] * np.exp(-kep * (t[i] - t[:i+1]))
+                ct[i] = ktrans * np.trapz(integrand, t[:i+1])
+        
+        return ct
+    
+    def fit(self, t: np.ndarray, ct: np.ndarray, cp: np.ndarray, 
```

Comment:
should be using external library `gpufit` (gpu accelerated c++ wrapper for levenberg-marquart fitting)